airbyte-webapp/src/views/Connector/ServiceForm/components/Controls/ConnectorServiceTypeControl/ConnectorServiceTypeControl.tsx

```diff
@@ -122,14 +124,18 @@ const ConnectorServiceTypeControl: React.FC<ConnectorServiceTypeControlProps> =
   const { formatMessage } = useIntl();
   const { openModal, closeModal } = useModalService();
   const { trackMenuOpen, trackNoOptionMessage, trackConnectorSelection } = useAnalyticsTrackFunctions(formType);
+  const reveal = useReveal();
 
   const workspace = useCurrentWorkspace();
+  const revealSort = useExperiment("connector.revealSort", true);
   const orderOverwrite = useExperiment("connector.orderOverwrite", {});
+
   const availableConnectorDefinitions = useAvailableConnectorDefinitions(availableServices, workspace);
-  const sortedDropDownData = useMemo(
+  const pinnedDropDownData = useMemo(
     () => getSortedDropdownDataUsingExperiment(availableConnectorDefinitions, orderOverwrite),
     [availableConnectorDefinitions, orderOverwrite]
   );
+  const sortedDropDownData = revealSort ? sortRevealSources(reveal?.tech, pinnedDropDownData) : pinnedDropDownData;
```

Comment:
`reveal?.tech` is `undefined` 

```suggestion
  const sortedDropDownData = revealSort ? sortRevealSources(reveal?.company.tech, pinnedDropDownData) : pinnedDropDownData;
```